kotlinx-coroutines-core/jvm/test/ThreadLocalsLeaksTest.kt

```diff
@@ -0,0 +1,95 @@
+package kotlinx.coroutines
+
+import kotlinx.coroutines.testing.TestBase
+import java.lang.ref.WeakReference
+import kotlin.coroutines.AbstractCoroutineContextElement
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.ContinuationInterceptor
+import kotlin.coroutines.CoroutineContext
+import kotlin.test.Test
+import kotlin.test.assertNull
+
+/*
+ * This is an adapted verion of test from #4296.
+ *
+ * qwwdfsad: the test relies on System.gc() actually collecting the garbage.
```

Comment:
Thanks for the comment, mate! That's indeed the issue, rewrote the tests to be GC-agnostic